rust/src/chain_crypto/algorithms/ed25519_extended.rs

```diff
@@ -0,0 +1,89 @@
+use crate::chain_crypto::key::{AsymmetricKey, AsymmetricPublicKey, SecretKeyError, SecretKeySizeStatic};
+use crate::chain_crypto::sign::SigningAlgorithm;
+
+use super::ed25519 as ei;
+
+use cryptoxide::ed25519;
+use rand_os::rand_core::{CryptoRng, RngCore};
+
+use ed25519_bip32::{XPrv, XPRV_SIZE};
+
+/// ED25519 Signing Algorithm with extended secret key
+pub struct Ed25519Extended;
+
+#[derive(Clone)]
+pub struct ExtendedPriv([u8; ed25519::PRIVATE_KEY_LENGTH]);
+
+impl AsRef<[u8]> for ExtendedPriv {
+    fn as_ref(&self) -> &[u8] {
+        &self.0[..]
+    }
+}
+
+impl ExtendedPriv {
+    pub fn from_xprv(xprv: &XPrv) -> Self {
+        let mut buf = [0; ed25519::PRIVATE_KEY_LENGTH];
+        xprv.get_extended_mut(&mut buf);
+        ExtendedPriv(buf)
+    }
+}
+
+impl AsymmetricKey for Ed25519Extended {
+    type Secret = ExtendedPriv;
+    type PubAlg = ei::Ed25519;
+
+    const SECRET_BECH32_HRP: &'static str = "ed25519e_sk";
+
+    fn generate<T: RngCore + CryptoRng>(mut rng: T) -> Self::Secret {
+        let mut priv_bytes = [0u8; XPRV_SIZE];
+        rng.fill_bytes(&mut priv_bytes);
+        let xprv = XPrv::normalize_bytes_force3rd(priv_bytes);
+
+        let mut out = [0u8; ed25519::PRIVATE_KEY_LENGTH];
+        xprv.get_extended_mut(&mut out);
```

Comment:
Why did this get changed from `get_extended` to `get_extended_mut`?

It seems this change originates from this upstream commit (https://github.com/input-output-hk/chain-libs/commit/0295b6db5fea512838a957739cc33746c6d0da88), and the code is only partially ported into this PR. I'm not sure if it's safe to replace just this part and not make the rest of the code changes from this commit.